rpcserver.go

```diff
@@ -1059,9 +1059,65 @@ func (r *rpcServer) ListAssets(ctx context.Context,
 			"and include_leased")
 	}
 
+	constraints := tapfreighter.CommitmentConstraints{
+		MinAmt:         req.MinAmount,
+		MaxAmt:         req.MaxAmount,
+		CoinSelectType: tapsend.ScriptTreesAllowed,
+	}
+
+	if req.GroupKey != nil {
```

Comment:
Should check the length here instead, which asserts `nil` as well as an empty slice.